shell.go

```diff
@@ -84,6 +84,42 @@ func NewShellWithClient(url string, c *gohttp.Client) *Shell {
 	}
 }
 
+// Get shell from environmental variables, default api path or gateway.
+func DefaultShell() (*Shell, error) {
```

Comment:
It would be nice to use a `sync.Once` and cache the default client. Thoughts?

Reply:
Will DefaultShell be called more than once? I think people will normally call it for one shot.

Reply:
I can see people calling `DefaultShell().DoThing()` quite often.